tests/unit/firebaseRef.spec.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+describe('firebaseRef', function () {
+
+  var firebaseRefProvider;
+  var MOCK_URL = 'https://stub.firebaseio-demo.com/'
+
+  beforeEach(module('firebase', function(_firebaseRefProvider_) {
+    firebaseRefProvider = _firebaseRefProvider_;
+  }));
+
+  describe('registerUrl', function() {
+
+    it('creates a single reference with a url', inject(function() {
+      firebaseRefProvider.registerUrl(MOCK_URL);
+      expect(firebaseRefProvider.$get()).toBeAFirebaseRef();
+    }));
+
+    it('creates a default reference with a config object', inject(function() {
+      firebaseRefProvider.registerUrl({
+        default: MOCK_URL
+      });
+      var firebaseRef = firebaseRefProvider.$get();
+      expect(firebaseRef).toBeAFirebaseRef();
+    }));
+
+    it('creates multiple references with a config object', inject(function() {
+      firebaseRefProvider.registerUrl({
+        default: MOCK_URL,
+        messages: MOCK_URL + 'messages'
+      });
+      var firebaseRef = firebaseRefProvider.$get();
+      expect(firebaseRef).toBeAFirebaseRef();
+      expect(firebaseRef.messages).toBeAFirebaseRef();
+    }));
+
+    it('should throw an error when no url is provided', inject(function () {
+      function errorWrapper() {
+        firebaseRefProvider.registerUrl();
+        firebaseRefProvider.$get();
+      }
+      expect(errorWrapper).toThrow();
+    }));
+
+    it('should throw an error when a reserved property is used', inject(function() {
+      function errorWrapper() {
+        firebaseRefProvider.registerUrl({
+          path: 'hello'
```

Comment:
I'd like to see a valid URI replace `'hello'`; I know this is nitpicky, but we're not checking what gets thrown, and if the URI is invalid and that causes a throw somewhere, we could end up with a false positive.